usvm-python/usvm-python-main/src/main/kotlin/org/usvm/machine/symbolicobjects/rendering/PyValueBuilder.kt

```diff
@@ -0,0 +1,364 @@
+package org.usvm.machine.symbolicobjects.rendering
+
+import io.ksmt.expr.KInt32NumExpr
+import io.ksmt.expr.KInterpretedValue
+import io.ksmt.sort.KIntSort
+import org.usvm.UConcreteHeapRef
+import org.usvm.api.typeStreamOf
+import org.usvm.isAllocatedConcreteHeapRef
+import org.usvm.isStaticHeapRef
+import org.usvm.machine.PyState
+import org.usvm.machine.interpreters.concrete.ConcretePythonInterpreter
+import org.usvm.machine.model.PyModelHolder
+import org.usvm.machine.symbolicobjects.InterpretedAllocatedOrStaticSymbolicPythonObject
+import org.usvm.machine.symbolicobjects.InterpretedInputSymbolicPythonObject
+import org.usvm.machine.symbolicobjects.InterpretedSymbolicPythonObject
+import org.usvm.machine.symbolicobjects.UninterpretedSymbolicPythonObject
+import org.usvm.machine.symbolicobjects.memory.FloatMinusInfinity
+import org.usvm.machine.symbolicobjects.memory.FloatNan
+import org.usvm.machine.symbolicobjects.memory.FloatNormalValue
+import org.usvm.machine.symbolicobjects.memory.FloatPlusInfinity
+import org.usvm.machine.symbolicobjects.memory.containsField
+import org.usvm.machine.symbolicobjects.memory.dictContainsInt
+import org.usvm.machine.symbolicobjects.memory.dictContainsRef
+import org.usvm.machine.symbolicobjects.memory.dictIsEmpty
+import org.usvm.machine.symbolicobjects.memory.getBoolContent
+import org.usvm.machine.symbolicobjects.memory.getFieldValue
+import org.usvm.machine.symbolicobjects.memory.getFloatContent
+import org.usvm.machine.symbolicobjects.memory.getIntContent
+import org.usvm.machine.symbolicobjects.memory.getSliceContent
+import org.usvm.machine.symbolicobjects.memory.readArrayElement
+import org.usvm.machine.symbolicobjects.memory.readArrayLength
+import org.usvm.machine.symbolicobjects.memory.readDictIntElement
+import org.usvm.machine.symbolicobjects.memory.readDictRefElement
+import org.usvm.machine.symbolicobjects.memory.setContainsInt
+import org.usvm.machine.symbolicobjects.memory.setContainsRef
+import org.usvm.machine.symbolicobjects.memory.setIsEmpty
+import org.usvm.machine.types.ConcretePythonType
+import org.usvm.machine.types.MockType
+import org.usvm.machine.utils.MAX_INPUT_ARRAY_LENGTH
+import org.usvm.mkSizeExpr
+import org.usvm.python.model.PyCompositeObject
+import org.usvm.python.model.PyIdentifier
+import org.usvm.python.model.PyMockObject
+import org.usvm.python.model.PyPrimitive
+import org.usvm.python.model.PyTupleObject
+import org.usvm.python.model.PyValue
+import org.usvm.types.first
+
+class PyValueBuilder(
+    var state: PyState,
+    private val modelHolder: PyModelHolder,
```

Comment:
`PyValueBuilder` may be used during `concolicRun`, for example, for mocks. As I said, `PyModel` is not constant during execution, but `PyModelHolder` is. Ideally, `PyValueBuilder` shouldn't have `state`, because the values should be fixed. Unfortunately, we need `state` for several operations on memory. For example, now `UModel` doesn't have full information on static objects.